src/util.ts

```diff
@@ -89,6 +89,30 @@ export function createOutputRetainingCompilerHost(
   }
 }
 
+/**
+ * Returns the declarations for the given decorator.
+ */
+export function getDecoratorDeclarations(decorator: ts.Decorator, typeChecker: ts.TypeChecker) {
+  // Walk down the expression to find the identifier of the decorator function.
+  let node: ts.Node = decorator;
+  while (node.kind !== ts.SyntaxKind.Identifier) {
+    switch (node.kind) {
+      case ts.SyntaxKind.Decorator:
+      case ts.SyntaxKind.CallExpression:
+        node = (node as ts.Decorator | ts.CallExpression).expression;
+        break;
+      default:
+        throw new Error();
```

Comment:
Could you or Evan clarify what the message that used to be in decorator-annotator.ts meant?

        // PropertyAccessExpression is intentionally missing here,
        // because the rest of the rewriter does not handle such
        // expressions.

Would this refer to decorators of the form `@foo.bar`?